lib/templates/gametemplate-moves.js

```diff
@@ -0,0 +1,301 @@
+
+    if (txmsg.game_id !== this.game.id) {
+      return 0;
+    }
+
+    if (tx_step == this.game.step.game) {
+      return 1;
+    }
+
+    if (this.game.step.players[player]) {
+      if (tx_step == this.game.step.players[player]) {
+        return 1;
+      }
+
+      // MAY 28 - player has an updated move, but not the next move
+      // When on earth does this actually occur?
+      // -- this is a great question -- perhaps this pre-dates getting futureMoves right?
+      //
+      if (tx_step >= this.game.step.players[player] && tx_step < this.game.step.game) {
+        return 1;
+      }
+    }
+
+    return 0;
+  }
+
+  addNextMove(gametx) {
+    let gametxmsg = gametx.returnMessage();
+
+    ////////////
+    // HALTED  -- a safety catch for future moves
+    ////////////
+    if (this.game.halted == 1 || this.gaming_active == 1 || this.initialize_game_run == 0) {
+      console.info(`Save as future move because halted (${this.game.halted}) or active (${this.gaming_active})`);
+      this.addFutureMove(gametx);
+      return;
+    }
+
+    //Update player's step value
+    this.game.step.players[gametx.transaction.from[0].add] = gametxmsg.step.game;
+
+    //And master game step value (if actually incremented)
+    if (gametxmsg.step.game > this.game.step.game) {
+      this.game.step.game = gametxmsg.step.game;
+    }
+
+    window.location.hash = this.app.browser.modifyHash(window.location.hash, {step: gametxmsg.step.game});
+
+    //console.info(`Add Next Move (${gametxmsg.step.game}) to QUEUE: ${JSON.stringify(gametxmsg.turn)}`);
+    //console.log(JSON.parse(JSON.stringify(gametxmsg)));
+
+    //
+    // OBSERVER MODE -
+    //
+    if (this.game.player == 0) {
+      if (this.browser_active) {
+        this.observerControls.showLastMoveButton();
+        this.observerControls.updateStep(this.game.step.game);
+      }
+      
+      this.observerControls.game_states.push(this.game_state_pre_move);
+      this.observerControls.game_moves.push(gametx);
+      //To avoid memory overflow for long games
+      if (this.observerControls.game_states.length > 100) {
+        this.observerControls.game_states.shift();
+      }
+      if (this.observerControls.game_moves.length > 100) {
+        this.observerControls.game_moves.shift();
+      }
+    }
+
+    if (this.game.queue) {
+      for (let i = 0; i < gametxmsg.turn.length; i++) {
+        this.game.queue.push(gametxmsg.turn[i]);
+      }
+
+      this.saveFutureMoves(this.game.id);
+      this.saveGame(this.game.id);
+      this.startQueue();
+    } else {
+      console.error("No queue in game engine");
+    }
+  }
+
+  addFutureMove(gametx) {
+    if (!this.game.future) {
+      this.game.future = [];
+    }
+
+    if (!this.game.future.includes(JSON.stringify(gametx.transaction))) {
+      
+      console.log("Future TX:" , JSON.parse(JSON.stringify(gametx.returnMessage())));
+      this.game.future.push(JSON.stringify(gametx.transaction));
```

Comment:
check fields in transaction

Reply:
what is to check here? Sorry didn't understand.

Reply:
Tx doesn't have a transaction field now. use toJson() perhaps?